feature/home/src/main/java/com/epmedu/animeal/home/presentation/ui/MapboxMap.kt

```diff
@@ -1,24 +1,48 @@
 package com.epmedu.animeal.home.presentation.ui
 
 import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.runtime.Composable
-import androidx.compose.runtime.remember
+import androidx.compose.runtime.*
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalLifecycleOwner
 import androidx.compose.ui.viewinterop.AndroidView
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.repeatOnLifecycle
+import com.epmedu.animeal.home.presentation.model.FeedingPointUi
 import com.epmedu.animeal.home.presentation.model.MapLocation
-import com.epmedu.animeal.home.presentation.viewmodel.HomeState
+import com.epmedu.animeal.home.presentation.viewmodel.HomeViewModel
 import com.mapbox.geojson.Point
 import com.mapbox.maps.CameraOptions
 import com.mapbox.maps.MapInitOptions
 import com.mapbox.maps.MapView
 import com.mapbox.maps.ResourceOptions
+import com.mapbox.maps.plugin.annotation.generated.OnPointAnnotationClickListener
+import com.mapbox.maps.plugin.annotation.generated.PointAnnotationManager
+import com.mapbox.maps.plugin.annotation.generated.PointAnnotationOptions
 import com.mapbox.maps.plugin.locationcomponent.location
 import com.mapbox.maps.plugin.scalebar.scalebar
 
 @Composable
-fun MapboxMap(state: HomeState) {
+fun MapboxMap(
+    homeViewModel: HomeViewModel,
+    onFeedingPointClick: (point: FeedingPointUi) -> Unit
+) {
+    val lifecycle = LocalLifecycleOwner.current.lifecycle
+    val state by homeViewModel.stateFlow.collectAsState()
     val mapBoxView = mapView(state.mapBoxPublicKey, state.mapBoxStyleUri)
+    val markerController by remember { mutableStateOf(MarkerController(mapBoxView)) }
+
+    LaunchedEffect(Unit) {
+        lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+            homeViewModel.getFeedingPointsFlow().collect { pointsState ->
```

Comment:
Why don't you pass the flow itself instead of passing the viewmodel all the way down?

Reply:
not actual now, will close comment